### PR TITLE
Update and Expand New Developer's Guide

### PR DESCRIPTION
**Pull request by nsryan2**
@katyhuff This update addresses the points you raised in issue #561, but does not detail downloading from source. I hope to address changes in the process necessary to install from the source on the installation page. It links, as before to the installation page. The update adds information about creating an environment (something I found useful in building the website and using PyNE because I previously installed Anaconda 3.7), updates formatting for more consistency, details considerations and methods to check the version of dependencies (a problem noted in many issues I've read), and adds additional links to coding resources.

### CODE REVIEW

File: docs/devsguide/new_dev_guide.rst [hunk 1]
```diff
@@ -1,133 +1,213 @@
...
+To Install Homebrew
+```````````````````
+Run the commands ::
+
+	$ ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+	$ brew doctor
+	$ brew tap brewsci/science
 
 ---------------------
 Installing on Windows
```
Comment (gonuke) [inline docs/devsguide/new_dev_guide.rst:40]: Has anyone actually build PyNE on Windows recently (ever?)

Comment (gonuke) [reply docs/devsguide/new_dev_guide.rst:40]: If yes, then super!!! if not, then I wonder if we should remove this section?

File: docs/devsguide/new_dev_guide.rst [hunk 2]
```diff
@@ -1,133 +1,213 @@
...
+developer, likely, you will not have all of the necessary components.
 
-==================================
-Installing from Source
-==================================
-NOTE: Might have to install pip, home-brew, and other programs to correctly 
-install.
-      Also, installing PyNE isn’t the same as cloning the repository.
+A particularly important detail to pay attention to is the version of
+each dependency. For example, Python is currently supported in conda
```
Comment (gonuke) [inline docs/devsguide/new_dev_guide.rst]: Do you mean "PyNE is currently supported in ...."?

File: docs/devsguide/new_dev_guide.rst [hunk 3]
```diff
@@ -1,133 +1,213 @@
...
+version of 3.7.\* will result in errors. Setting your version to be
+compatible before you begin will help ensure you are successful.
 
-PyNE's `website <http://pyne.io/install/index.html>`__ will lead you through the 
-installation. 
 
-To contribute to the project, 
+-----------------------
+Creating an Environment
+-----------------------
```
Comment (gonuke) [inline docs/devsguide/new_dev_guide.rst:107]: This is a great addition! Thanks @nsryan2

File: docs/devsguide/new_dev_guide.rst [hunk 4]
```diff
@@ -1,133 +1,213 @@
...
+documentation can be found
+`here
+<https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-
+environments.html#sharing-an-environment>`__.
+
+
+==========================
+Signing up for list hosts:
+==========================
+Writing effective code isn’t easy. Thankfully, the PyNE developers can always be
```
Comment (gonuke) [inline docs/devsguide/new_dev_guide.rst]: I know you didn't write this, but upon review, maybe this sets a better tone:
```suggestion
Everyone faces challenges sometimes when writing effective code. Thankfully, the PyNE developers can always be 
```

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks for this big improvement @nsryan2!

I have a couple of comments, some based on things you inherited here.

**Comment by gonuke** [comment]
You'll want to add a news file as well. There is a template at `news/TEMPLATE.rst`.  (Maybe we should add that to the dev guide somewhere, as well?)

File: docs/devsguide/new_dev_guide.rst (hunk 1 above)
Comment (nsryan2) [reply docs/devsguide/new_dev_guide.rst:40]: I have not built it on Windows, but on Mac OS I found this step necessary.

Comment (gonuke) [reply docs/devsguide/new_dev_guide.rst:40]: Roger that.  Maybe we'll make an issue for someone to try it in a Windows conda environment (if we can find one).

**Comment by nsryan2** [comment]
> You'll want to add a news file as well. There is a template at `news/TEMPLATE.rst`. (Maybe we should add that to the dev guide somewhere, as well?)

I think it would be a really good idea to add to the guide, I had no idea until you brought it up. As a point of clarification this file should be named after the update to the document, and it should go in the news file?

**Comment by gonuke** [comment]
> I think it would be a really good idea to add to the guide, I had no idea until you brought it up. As a point of clarification this file should be named after the update to the document, and it should go in the news file?

Sorry about that - we should have fixed this already since it's been raised before.  Thanks for your patience.

File: docs/devsguide/new_dev_guide.rst (hunk 2 above)
Comment (nsryan2) [reply docs/devsguide/new_dev_guide.rst]: Yes, that was my intention. Thank you for the note

**Comment by gonuke** [comment]
This LGTM! - thanks @nsryan2

**Comment by gonuke** [comment]
I'll merge after testing, but might ask @bam241 to take care of pushing an update to the website....

**Review by katyhuff — APPROVED** [review]
Looks good to me too. Thanks @nsryan2 (and @gonuke for the the review and @bam241 for the website rebuild!)

**Review by gonuke — APPROVED** [review]

**Pull request merged**